design/one-pager-managed-resource-api-design.md

```diff
@@ -115,13 +118,24 @@ Related to https://github.com/crossplaneio/crossplane/issues/624
 The decision for an external resource name is made by the controller of that managed resource in `Create` phase.
 Possible cases are as following:
 * The provider doesn't allow us to specify a name. Fetch it after its client's `POST` call. Override `crossplane.io/external-name`
-annotation's value with what you get as name.
+  annotation's value with what you get as name.
 * The provider allows to specify a name.
   * If `crossplane.io/external-name` annotation has a value use it. If it's empty;
   * Use managed resource's name and write that into `crossplane.io/external-name` annotation.
 
 Use the value of that annotation as external resource name in _all_ queries.
 
+If the external resource supports labelling, we should label it with the managed
+resource name. While this is helpful for services with non-deterministic naming,
```

Comment:
Are you thinking that we add this label automatically? So if a user supplies labels we append this one to the set in the controller?

Reply:
> Are you thinking that we add this label automatically?

The infra stack developer would have to make sure they add the label in `ExternalClient` functions because the field to add the label differs.

> So if a user supplies labels we append this one to the set in the controller?

Yes